### ISSUE TITLE
Add resource and edge to new resource in transaction and crash site

### ISSUE THREAD

**Issue by mmzeeman** [zotonic/zotonic#1861]
When you add a resource and an edge to that new resource inside a transaction you will crash the site.

The reason for this is that the resource will be pivoted separate from the transaction inserting the resource and the edge.

Upon adding the edge the data of the not yet available resource will be retrieved. 

This line will get the id of the new resource, but assoc_props_row will return undefined. This value will be sent into the pivot_rsc_data fold leading to a crash of the pivot server, which will crash the site.

https://github.com/zotonic/zotonic/blob/0.x/src/support/z_pivot_rsc.erl#L652

```erlang
gen_server 'z_pivot_rsc$channelwww' terminated with reason: no function clause matching proplists:get_value(language, undefined, undefined) line 226
```

I'm not sure why this suddenly happens, because this used to work fine. 

Should this be fixed by filtering `undefined` responses from assoc_props_row, or should the pivot be delayed until the transaction completes?

**Comment by mmzeeman** [comment]
Related to this? https://github.com/zotonic/zotonic/commit/ebea4524432755943a12872e0ea78bc432e6637e

**Comment by mmzeeman** [comment]
Adding the option `[no_touch]` to the `m_edge:insert` skips the pivot operation and prevents the crash.

How should we fix this? Add a case statement to this function filtering out `undefined` responses?

https://github.com/zotonic/zotonic/blob/0.x/src/support/z_pivot_rsc.erl#L652

**Comment by mworrell** [comment]
Yes, as there are always race conditions we should not crash if a non-existing resource happens to be pivoted.

Either because it is not visible from the pivot-transaction, or because it doesn't exist at all.

If it was still in another transaction, then there will be a pivot-queue entry created, which will ensure a later pivot.

**Comment by ddeboer** [comment]
This should be fixed in c986f7434584808316ac442c2f8ca4a9344a59f1.
